hips/hip-0019.md

```diff
@@ -0,0 +1,61 @@
+--- 
+hip: 0019
+title: "Add Exclude File Option to Helm Lint Command"
+authors: Danilo Patrucco
+created: "2024-07-03"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+This proposal introduces enhancements to the `helm lint` command, allowing the exclusion of specific files or patterns through a `.helmlintconfig.yaml` file, analogous to `.gitignore` or `.helmignore`. Additionally, a `--lint-config-file` flag will be added to specify alternative locations for the `.helmlintconfig.yaml` file. This feature is particularly useful in projects with multiple sub-charts. An environment variable `HELM_LINT_CONFIG_FILE` will also be introduced to mirror the functionality of the flag.
+
+## Motivation
+In large Helm charts, certain files or configurations may not conform to standard linting rules but are accepted by the maintainers. The current `helm lint` process evaluates all files within a chart, often leading to irrelevant warnings or errors. An option to exclude specific files and the ability to specify config file locations would streamline the linting process.
+
+## Rationale
+The aim of this proposal is to grant developers and integrators more control over the linting process by enabling:
+- Exclusions for files imported or managed by third parties that are not directly editable.
+- Exclusions for files that may not adhere to static linting rules.
+- Exclusions for files containing intentional deviations from standard practices due to specific deployment scenarios.
+
+The `.helmlintconfig.yaml` file will support simple patterns for matching files and directories, enhancing ease of management. The `--lint-config-file` flag will facilitate centralized management of lint exclusions in complex projects.
+
+## Specification
+### `.helmlintconfig.yaml` File Format
+The `.helmlintconfig.yaml` file enables chart developers to specify filenames or glob patterns to exclude from linting. The format is YAML, with patterns specified under a `lintIgnore` key, making future extensions for additional configurations feasible.
+
+#### Example
+```
+# .helmlintconfig.yaml file example
+lintIgnore:
```

Comment:
perhaps a format more like (bikeshedding hard here):

```
# .helmlintconfig.yaml file example
pathIgnore:
- "migrations/templates/job.yaml"
- "gitlab/templates/shared-secrets/self-signed-cert-job.yml"
- "gitlab/templates/*.yaml"
errorIgnore:
- "chart metadata is missing these dependencies*"
- "<include 'fullname' .>"
```

Reply:
I modified this section a little, but kept it similar to our suggestion because an error (the same error specifically) can occur in multiple charts / sub-charts, and connecting it to a file path would be beneficial.